### DISCUSSION TITLE
Serverless Function Size Limit exceeded in Gatsby

### DISCUSSION THREAD

**Discussion by website-sevdesk** [vercel/community#2876] [Help]
### Summary

Hey there, 
we are using Gatsby with its SSR function getServerData.
In there we fetch the page data from datocms and want to display it. Locally it works fine but in production the build fails while updating the pages. The Serverless Function Size is exceeded. Its 55MB instead of 50MB.
Do you have any idea how to display the dependencies of the function via vercel cli and how to reduce the size.
Ive tried a lot of this one https://github.com/orgs/vercel/discussions/103, but nothing works for me.
I need to figure out which dependencies are in there but i dont know the command for that.
Thank you very much!

### Steps to Reproduce

Error: The Serverless Function "[slug]/index.html" is 55.68mb which exceeds the maximum size limit of 50mb. Learn More: https://vercel.link/serverless-function-size
--
08:14:55.086 | NOW_SANDBOX_WORKER_MAX_LAMBDA_SIZE: The Serverless Function "[slug]/index.html" is 55.68mb which exceeds the maximum size limit of 50mb.

**Comment by amyegan** [comment]
I usually use tools like [bundlejs.com](https://bundlejs.com/) and [bundlephobia.com](https://bundlephobia.com/) to help me figure out which dependencies to modify or remove.

**Reply by website-sevdesk** [reply]
thank you so much, but it doesnt help for the given issue, all the data which gets rendered in buildtime (gatsby-node.js) has impact on the serverless function size

**Comment by website-sevdesk** [comment]
I have the this code in the vercel.json:
` "functions": {
    "src/templates/customTemplate.js": {
      "excludeFiles": "{gatsby-node.js,node_modules/**/*,.gatsby/**/*,.vercel/**/*}"
    }
  },`
  
  i tried so much in there but nothing has any effect on the size, i dont know if this is working in gatbsy

**Reply by natapg** [reply]
any news on this? I'm trying to add ssr to my website, and even with a simple function that reads cookies and returns a string, the weight of the function is 44mb, and sometimes it takes 4 seconds to execute (I think because of the cold start) and it doesn't I find a way to debug it
